s3stream/src/test/java/com/automq/stream/s3/compact/CompactionManagerTest.java

```diff
@@ -563,7 +566,7 @@ public void testCompactionShutdown() throws Throwable {
 
         objectManager = Mockito.spy(MemoryMetadataManager.class);
         objectStorage = Mockito.spy(MemoryObjectStorage.class);
-        List<Pair<InvocationOnMock, CompletableFuture<ByteBuf>>> invocations = new ArrayList<>();
+        final BlockingQueue<Pair<InvocationOnMock, CompletableFuture<ByteBuf>>> invocations = new LinkedBlockingQueue<>();
```

Comment:
this fix is for flaky unit test of testCompactionShutdown the origin invocations is not thread safe.